x-pack/plugins/case/server/client/configure/get_mappings.ts

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { ConnectorMappingsAttributes, ConnectorTypes } from '../../../common/api';
+import { CaseClientFactoryArguments, MappingsClient } from '../types';
+// eslint-disable-next-line @kbn/eslint/no-restricted-paths
+import { ACTION_SAVED_OBJECT_TYPE } from '../../../../actions/server/saved_objects';
+
+export const getMappings = ({
+  savedObjectsClient,
+  connectorMappingsService,
+}: CaseClientFactoryArguments) => async ({
+  actionsClient,
+  caseClient,
```

Comment:
I think there is no need to pass the caseClient. You can import the getFields as

```
import { getFields as getFieldsFactory } from './get_fields.ts'

```

and use it as 

```
const getFields = getFieldsFactory({ savedObjectsClient, connectorMappingsService })
const res = await getFields({
      actionsClient,
      connectorId,
      connectorType,
    });
```

Reply:
Although, I am not sure if this will create memory issues on multiple requests as it will create a new function on each one. @XavierM what do you think about it?

Reply:
paired with @XavierM on this part so pretty sure it should be here.